src/efs_parser/BaseFile.py

```diff
@@ -38,7 +41,9 @@ class BaseFile(object):
 
     def __init__(self, file):
         self.file = file
-        self.local = self.is_local(file)
+        self.bucketname = "encode-public"   # Default will be overwritten by split_s3_components
+        self.region_name = "us-west-2"      # Default will be overwritten by split_s3_components
```

Comment:
defaults should be `None` not set to a bucket and region

Reply:
also why doesn't the constructor split the `s3url`  so we report an exception early on if the url is not in the format we require

Reply:
if the split happens in the constructor, that avoids setting default